release/release.bzl

```diff
@@ -0,0 +1,68 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+def _reconstruct_label(label):
+    if label.workspace_root:
+        return "@{}//{}:{}".format(label.workspace_root, label.package, label.name)
+    else:
+        return "//{}:{}".format(label.package, label.name)
+
+def _release_impl(ctx):
+    label = _reconstruct_label(ctx.label)
+    artifacts = []
+    runfiles = []
+    for k, v in ctx.attr.artifacts.items():
+        files = k[DefaultInfo].files.to_list()
+        if len(files) > 1:
```

Comment:
In a follow up PR to add support for targets that export multiple files.

Reply:
We can investigate it.  The current rationale for only supporting 1 file per target is so the targets represent an `{artifact: "Text description, ...}` mapping for files uploaded to github.

You can release multiple artifacts at once - the CRT repo does this to release, as separate binaries, the cc65 compiler, the win32 compiler, the win64 compiler and the qemu simulators.

Lastly, you're meant to compose this together with the existing bazel packaging rules if you want to create an archive of release artifacts with a certain file structure inside the archive.  You can see an example of this in the packaging rules in the opentitan repo (start with the `pkg_tar` rule in `//release:BUILD`).

Reply:
I tried to include `//: provisioning_appliance_containers` from the `ot-provisioning` repo into a `pkg_tar`, but wasn't able to make it work. I didn't look close into it to try to understand why it didn't work. 

I agree, it probably makes more sense to require sub packaging.

Reply:
The reason for that appears to be the structure of the providers returned by the `container_bundle` rule:

The `container_bundle` rule is `executable=True`.  If I'm reading the source correctly, this is so you can do something like `bazel run //:my-bundle` and get an interactive docker session with your bundle (presumably to play around or manually check that you've got everything you wanted in the container image).

Executables have to provide the file to execute via the `DefaultInfo` provider's `executable` field and the rest of the needed files via `files` and `runfiles`.  This means that if you `bazel build` a container_bundle, the `DefaultInfo` is what you need to run docker in interactive mode.

The `container_bundle` rule _also_ provides a `BundleInfo` provider (which can be consumed by `container_push`) and an `OutputGroupInfo` provider, which contains the assembled `.tar` file.

In order to use the assembled tar file as a release artifact, we need to convince the releasing rule to choose that file.  I think we can do that with a `filegroup` rule which will will select the files from one of the output groups:
```
filegroup(
    name = "provisioning_appliance_containers_tar",
    srcs = [":provisining_appliance_containers"],
    output_group = "tar",
)
```

You should then be able to use the filegroup label in the `release` rule as an artifact.